docs/self-managed/setup/deploy/amazon/amazon-eks/terraform-setup.md

```diff
@@ -250,65 +252,426 @@ access_entries = {
+```
 
-output "postgres_endpoint" {
-  value = module.postgresql.aurora_endpoint
-  description = "The Postgres endpoint URL"
-}
+  </TabItem>
+  <TabItem value="irsa" label="IRSA">
+
+In addition to using standard username and password authentication, you can opt to use [**IRSA (IAM Roles for Service Accounts)**](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/) for secure, role-based access to your Aurora database. This method allows your EKS workloads to assume IAM roles without needing to manage AWS credentials directly.
+
+**Note**: Using IRSA is optional. If preferred, you can continue using traditional password-based authentication for database access.
+
+If you choose to use IRSA, you’ll need to take note of the **IAM role** created for Aurora and the **AWS Account ID**, as these will be used later to annotate the Kubernetes service account.
+
+##### Aurora IRSA role and policy
+
+The Aurora module uses outputs from the EKS cluster module to configure the IRSA role and policy. Below are the required parameters:
+
+Here’s how to define the IAM role trust policy and access policy for Aurora:
+
+```hcl reference
+https://github.com/camunda/camunda-tf-eks-module/blob/feature/opensearch-doc/examples/camunda-8.6-irsa/db.tf
+```
+
+Once the IRSA configuration is complete, ensure you **record the IAM role name** (from the `iam_aurora_role_name` configuration), it is required to annotate the Kubernetes service account in the next step.
+
+</TabItem>
+</Tabs>
+
+#### Step 3: additional customization
+
+You can further customize the Aurora cluster setup through various input options. Refer to the [Aurora module documentation](https://github.com/camunda/camunda-tf-eks-module/blob/2.6.0/modules/aurora/README.md) for more details on other customization options.
+
+### OpenSearch module setup
+
+:::info Optional module
+
+If you don't want to use this module, you can skip this section.
+However, please note that you may need to adjust the following instructions to remove references to this module.
+
+If you choose not to use this module, you'll need to either provide a managed Elasticsearch or OpenSearch service or use the internal deployment by the Camunda Helm chart in Kubernetes.
+:::
+
+The OpenSearch module creates an OpenSearch domain intended for Camunda platform. OpenSearch is a powerful alternative to Elasticsearch. For more information on using OpenSearch with Camunda, refer to the [Camunda documentation](./self-managed/setup/guides/using-existing-opensearch/).
+
+:::note Available since Camunda 8.4
+
+As of the 8.4 release, Zeebe, Operate, and Tasklist are now compatible with [Amazon OpenSearch Service](https://aws.amazon.com/de/opensearch-service/) 2.5.x. Note that using Amazon OpenSearch Service requires [setting up a new Camunda installation](/self-managed/setup/overview.md). A migration from previous versions or Elasticsearch environments is currently not supported.
+
+:::
```

Comment:
same comment as in the eksctl doc on whether the note still makes sense or whether it should be just be simplified to not being able to mix and match the chosen "database".

It's a `next` version, where all components are compatible (some more than others ^^).